lib/server/monalisa-synchronization/MonAlisaSynchronizer.js

```diff
@@ -71,13 +71,22 @@ class MonAlisaSynchronizer {
                 }
             };
 
+            const errors = {};
             for (const monalisaDataPass of dataPasses) {
                 await dataSource.transaction(() => transactionablePipeline(monalisaDataPass))
-                    .catch((error) => this.logger.errorMessage(error.message, {}));
+                    .catch((error) => {
+                        if (!errors[error.message]) {
+                            errors[error.message] = [];
+                        }
+                        errors[error.message].push(monalisaDataPass.name);
+                    });
+            }
+            if (Object.entries(errors).length !== 0) {
+                this.logger.errorMessage(JSON.stringify(errors), {});
             }
```

Comment:
I think this is a strange way to proceed. If you have an error message very long, like an HTML error page, the error message you will display will be unreadable.
I would rather do it the following way:
Loop over the error messages, and put one log like: `Error occurred for data passes ${dataPasses.join(',')}:\n${message}`
Doing so, I would rather use a map than an object...

Reply:
refactored, used your formatting. and the longest expected error there is `Unexpected token '<', "<!doctype "... is not valid JSON`